### PR TITLE
Feature/offline missed out features

### PR DESCRIPTION
**Pull request by parunagiri**
Hi @backer ,

I have add all missed offline feature. Please do review and let me your thoughts on these below items

   1. Pause, Resume and Cancel download feature added
   2. DRM licence rental/purchase flow added
   3. onError callback method added report player error back to react native
   4. authToken has been added all edge api calls

### CODE REVIEW

File: android/src/main/java/jp/manse/BrightcovePlayerUtil.java [hunk 1]
```diff
@@ -56,6 +64,45 @@ public void getOfflineVideoStatuses(String accountId, String policyKey, Promise
         account.getOfflineVideoStatuses(promise);
     }
 
+    /**
+     * [getVideoIdFromOfflineToken] will act as dummy method to support iOS videoToken feature
```
[the remark this answers is not in the document]
Comment (parunagiri) [reply android/src/main/java/jp/manse/BrightcovePlayerUtil.java:68]: Yes @backer . This is not actually dummy method. It resolved purpose to support both platform

File: ios/BrightcovePlayer.m [hunk 2]
```diff
@@ -81,23 +86,66 @@ -(void) setupVideoProperties: (BCOVVideo*) video {
   return [BCOVPlayerSDKManager.sharedManager createPlaybackControllerWithSessionProvider:provider viewStrategy:nil];
 }
 
+- (void)createNewPlaybackController {
+  // The playback controller, as well as the download manager will work with either FairPlay-protected
```
[the remark this answers is not in the document]
Comment (PratheeshDBennet) [reply ios/BrightcovePlayer.m:90]: Hi @backer. Thanks for reviewing. Yes, you are right, this is the native side implementation of the Fairplay configurations for iOS. Please do correct me if I am missing out on anything on your question.

File: android/src/main/java/jp/manse/OfflineVideoDownloadSession.java [hunk 3]
```diff
@@ -6,61 +6,112 @@
...
+        this.offlineCatalog = new OfflineCatalog.Builder(context, DefaultEventEmitter.sharedEventEmitter, accountId)
+                .setPolicy(policyKey)
+                .build();
         this.offlineCatalog.setMeteredDownloadAllowed(true);
         this.offlineCatalog.setMobileDownloadAllowed(true);
         this.offlineCatalog.setRoamingDownloadAllowed(true);
         this.offlineCatalog.addDownloadEventListener(this);
         this.listener = listener;
-    }
+    }    private final EventListener licenseEventListener = event -> {
```
[the remark this answers is not in the document]
Comment (parunagiri) [reply android/src/main/java/jp/manse/OfflineVideoDownloadSession.java:65]: Sure. Thank you

File: android/src/main/java/jp/manse/BrightcovePlayerAccount.java [hunk 4]
```diff
@@ -238,11 +318,27 @@ private NativeArray collectNativeOfflineVideoStatuses() {
             map.putString(CALLBACK_KEY_VIDEO_ID, session.videoId);
             map.putString(CALLBACK_KEY_VIDEO_TOKEN, session.videoId);
             map.putDouble(CALLBACK_KEY_DOWNLOAD_PROGRESS, session.downloadProgress);
+            map.putString(CALLBACK_KEY_DOWNLOAD_STATE, getOfflineSessionStatus(session));
             statuses.pushMap(map);
         }
         return statuses;
     }
 
+    private String getOfflineSessionStatus(OfflineVideoDownloadSession session) {
```
[the remark this answers is not in the document]
Comment (parunagiri) [reply android/src/main/java/jp/manse/BrightcovePlayerAccount.java:327]: I missed the queue state. I will add. Thank you

**Review by backer — APPROVED** [review]
thanks for updating! 👍

File: ios/BrightcovePlayer.m (hunk 2 above)
[the remark this answers is not in the document]
Comment (PratheeshDBennet) [reply ios/BrightcovePlayer.m:90]: Hi @backer. Sure no problem, totally understood :)

**Review by GersonM — APPROVED** [review]

**Pull request closed without merging**